flink/v1.20/flink/src/main/java/org/apache/iceberg/flink/data/ParquetWithFlinkSchemaVisitor.java

```diff
@@ -51,106 +51,100 @@ public static <T> T visit(
-            try {
-              T elementResult = null;
-              if (repeatedElement.getFieldCount() > 0) {
-                elementResult = visitField(element, repeatedElement.getType(0), visitor);
-              }
-
-              return visitor.list(array, group, elementResult);
-
-            } finally {
-              visitor.fieldNames.pop();
-            }
-
-          case MAP:
-            Preconditions.checkArgument(
-                !group.isRepetition(Type.Repetition.REPEATED),
-                "Invalid map: top-level group is repeated: %s",
-                group);
-            Preconditions.checkArgument(
-                group.getFieldCount() == 1,
-                "Invalid map: does not contain single repeated field: %s",
-                group);
-
-            GroupType repeatedKeyValue = group.getType(0).asGroupType();
-            Preconditions.checkArgument(
-                repeatedKeyValue.isRepetition(Type.Repetition.REPEATED),
-                "Invalid map: inner group is not repeated");
-            Preconditions.checkArgument(
-                repeatedKeyValue.getFieldCount() <= 2,
-                "Invalid map: repeated group does not have 2 fields");
-
-            Preconditions.checkArgument(
-                sType instanceof MapType, "Invalid map: %s is not a map", sType);
-            MapType map = (MapType) sType;
-            RowField keyField =
-                new RowField("key", map.getKeyType(), "key of " + map.asSummaryString());
-            RowField valueField =
-                new RowField("value", map.getValueType(), "value of " + map.asSummaryString());
-
-            visitor.fieldNames.push(repeatedKeyValue.getName());
-            try {
-              T keyResult = null;
-              T valueResult = null;
-              switch (repeatedKeyValue.getFieldCount()) {
-                case 2:
-                  // if there are 2 fields, both key and value are projected
-                  keyResult = visitField(keyField, repeatedKeyValue.getType(0), visitor);
-                  valueResult = visitField(valueField, repeatedKeyValue.getType(1), visitor);
-                  break;
-                case 1:
-                  // if there is just one, use the name to determine what it is
-                  Type keyOrValue = repeatedKeyValue.getType(0);
-                  if (keyOrValue.getName().equalsIgnoreCase("key")) {
-                    keyResult = visitField(keyField, keyOrValue, visitor);
-                    // value result remains null
-                  } else {
-                    valueResult = visitField(valueField, keyOrValue, visitor);
-                    // key result remains null
-                  }
-                  break;
-                default:
-                  // both results will remain null
+      LogicalTypeAnnotation annotation = group.getLogicalTypeAnnotation();
+      if (LogicalTypeAnnotation.listType().equals(annotation)) {
```

Comment:
Nit: we have two approaches here, in [GenericArrowVectorAccessorFactory.java](https://github.com/apache/iceberg/pull/11874/files#diff-7f5d0c13cd8666c832fddaa6fa96191125a536e9e60fede58abb87d675da582e) we use `instanceof` and here we use `.equals`. Maybe good to use one way or the other. I have a slight preference for `instanceof` since it reads from left to right :)

Reply:
@Fokko Thanks for your suggestion, changed to `instanceof`.